lightningd/log.c

```diff
@@ -282,10 +288,21 @@ void log_io(struct log *log, enum log_level dir,
 
 	assert(dir == LOG_IO_IN || dir == LOG_IO_OUT);
 
+	/* Print first, in case we need to truncate. */
+	if (l->level >= log->lr->print_level)
+		log->lr->print(log->prefix, l->level, false,
+			       &l->time, str,
+			       data, len, log->lr->print_arg);
+
 	l->log = tal_strdup(l, str);
+
+	/* Don't immediately fill buffer with giant IOs */
+	if (len > log->lr->max_mem / 64) {
```

Comment:
Sorry, I have a question here: under what case is the io date so giant?...I can't catch it.

Reply:
log_io of JSON, where JSON is a listchannel command for a million channels :)

Reply:
It also triggers far earlier than 1m channels, each call to `listchannels` on mainnet will cause a number of log cleanups currently.